cpp/src/arrow/array/builder_nested.cc

```diff
@@ -72,6 +78,17 @@ Status ListBuilder::Append(bool is_valid) {
   return AppendNextOffset();
 }
 
+Status ListBuilder::AppendNulls(int64_t length) {
+  RETURN_NOT_OK(Reserve(length));
+  RETURN_NOT_OK(CheckNextOffset());
+  UnsafeAppendToBitmap(length, false);
+  const int64_t num_values = value_builder_->length();
+  for (int64_t i = 0; i < length; ++i) {
```

Comment:
memcpy instead of the whole loop?

Reply:
Hmm, how would you use `memcpy` here?

Reply:
It's unclear how much of an optimization it would be to malloc a piece of memory, copy values into it, then do a vector append to `offsets_builder_`, probably not much